### ISSUE TITLE
TfIdfSearchIndex.indexDocument breaks when encountering word "constructor"

### ISSUE THREAD

**Issue by licyeus** [bvaughn/js-search#15]
In version `1.3.2`:

```
const search = new JsSearch.Search('foo')
search.addIndex('foo')
search.addDocuments([{ foo: 'constructor' }])
```

throws `Uncaught TypeError: Cannot read property 'constructor' of undefined` inside `TfIdfSearchIndex.indexDocument`.

Thanks!

**Comment by bvaughn** [comment]
Hey...that's a fun bug. I'm using vanilla `Object` to map search indexes and `someObject.constructor` is reserved for the...constructor. I'd rather use `Map` but that would involve raising my supported browsers to IE11+, which would require a major version bump at least. Hm.

By the way, it doesn't make sense to add a search index on the same field as your UID field. (UIDs must always be a direct match, meaning that if you ever wanted to lookup an object by the UID, you'd just do `collection['your-uid']` or `collection.yourUID`.) I assume you know this though and were just providing a rough example.

**Comment by bvaughn** [comment]
So the reason this is occurring is due to a specific index, implementation, `TfIdfSearchIndex`. It stores metadata for a given search token (eg "constructor" in this case) in an internal object so that it can track the number of times it occurs within a given document and within the entire corpus. This information is later used to sort search results by a priority.

The index checks for each new token to determine if it needs to create a new metadata map _or_ increment a previous one. In this case, it returns a false-positive lookup for "constructor". I _think_ the fix is just to add constraints to the check. I should be able to roll a fix out shortly.

**Comment by bvaughn** [comment]
Fixed in version 1.3.3! Thanks for the report and repro case. 😄

**Comment by licyeus** [comment]
Awesome. Thanks much for the quick turn around! It's working as expected now. :tada:
